chalk-ir/src/lib.rs

```diff
@@ -1194,11 +1219,38 @@ impl<T> Binders<T> {
         }
     }
 
+    /// Unwraps and returns the value within, but only if it contains
+    /// no vars bound by this binder. (In other words, if this binder --
+    /// and indeed any enclosing binder -- doesn't bind anything at
+    /// all.) Otherwise, returns `None`.
+    ///
+    /// (One could imagine having a method that just unwraps a single
+    /// binder, but permits late-bound vars bound by enclosing
+    /// binders, but that would require adjusting the debruijn
+    /// indices, and given the shallow binding structure we often use,
+    /// would not be that useful.)
+    pub fn no_bound_vars<I: Interner>(&self, interner: &I) -> Option<&T>
```

Comment:
Do we actually use this anywhere? I don't see it. If not, let's just not add it. I know rustc has it, but I'm not sure that's a valid reason on its own. Also remember to update the doc comment for skip_binders to not mention this.